pkg/plugins/resources/temurin/api.go

```diff
@@ -0,0 +1,282 @@
+		return result, err
+	}
+
+	return result, nil
+}
+
+func (t Temurin) apiParseVersion(version string) (result parsedVersion, err error) {
+	apiEndpoint := fmt.Sprintf(
+		"%s/%s",
+		parseVersionEndpoint,
+		version,
+	)
+
+	body, err := t.apiGetBody(apiEndpoint)
+	if err != nil {
+		logrus.Errorf("something went wrong while parsing the version %q with the Temurin API available operating systems %q\n", version, err)
+		return result, err
+	}
+
+	err = json.Unmarshal(body, &result)
+	if err != nil {
+		logrus.Errorf("something went wrong while decoding response from Temurin API %q\n", err)
+		return result, err
+	}
+
+	return result, nil
+}
+
+func (t Temurin) apiGetReleaseName() (result string, err error) {
+	var versionRange string
+
+	// If user specified a custom version, we have to normalize and validate it
+	if t.spec.SpecificVersion != "" {
+		parsedVersion, err := t.apiParseVersion(t.spec.SpecificVersion)
+		if err != nil {
+			return "", err
+		}
+
+		versionRange = fmt.Sprintf("(%d.%d.%d, %d.%d.%d]",
+			parsedVersion.Major,
+			parsedVersion.Minor,
+			parsedVersion.Security,
+			parsedVersion.Major,
+			parsedVersion.Minor,
+			parsedVersion.Security+1,
+		)
+
+	} else {
+		featureVersion := t.spec.FeatureVersion
+		if featureVersion == 0 {
+			featureVersion, err = t.apiGetLastFeatureRelease()
+			if err != nil {
+				return "", err
+			}
+		}
+
+		versionRange = fmt.Sprintf("(%d.0.0, %d.0.0]", featureVersion, featureVersion+1)
+	}
+
+	apiEndpoint := fmt.Sprintf(
+		"%s?heap_size=normal&image_type=%s&page=0&page_size=10&project=jdk&release_type=%s&semver=true&sort_method=DEFAULT&sort_order=DESC&vendor=eclipse&version=%s",
+		releaseNamesEndpoint,
+		t.spec.ImageType,
+		t.spec.ReleaseType,
+		// Mandatory URL encoding otherwise empty responses or HTTP errors
+		url.QueryEscape(versionRange),
+	)
+
+	logrus.Debugf("[temurin] using API endpoint %q", apiEndpoint)
+
+	body, err := t.apiGetBody(apiEndpoint)
+	if err != nil {
+		logrus.Errorf("something went wrong while getting Temurin API latest release information %q\n", err)
+		return result, err
+	}
+
+	var apiResult releaseInformation
+	err = json.Unmarshal(body, &apiResult)
+	if err != nil {
+		logrus.Debugf("[temurin] Failed decoding the response: %q\n", err)
+		return result, fmt.Errorf("[temurin] No release found matching provided criteria. Use '--debug' to get details.")
+	}
+
+	if len(apiResult.Releases) == 0 {
+		logrus.Debug("[temurin] empty response for 'release_names'.")
+		return result, fmt.Errorf("[temurin] No release found matching provided criteria. Use '--debug' to get details.")
+	}
+
+	// Return only the most recent, e.g. the first one (sort is DESC in the URL)
+	return apiResult.Releases[0], nil
+}
+
+func (t Temurin) apiGetInstallerUrl(releaseName string) (result string, err error) {
+	apiEndpoint := fmt.Sprintf(
+		"%s/%s/%s/%s/%s/hotspot/normal/eclipse?project=jdk",
```

Comment:
Stupid question, is it normal that "jdk" is hardcoded?
Would it make sense to use jre or sbom?

Reply:
Differently asked, could we configure different project?

Reply:
Good question. I went for `jdk` as it's the only project which returned HTT/200 answers in my tests.

But there are other values available: worth adding them in the spec

From https://api.adoptium.net/q/swagger-ui/#/Binary/getBinaryByVersion: 

<img width="592" alt="Capture d’écran 2024-08-26 à 12 38 29" src="https://github.com/user-attachments/assets/2edc9f7d-41b4-4e0c-8ea5-2163d5744409">

Reply:
Updated the PR with the support for "Project" attribute